In [1]:
import pandas as pd

In [2]:
dataset=pd.read_csv("Placement.csv")
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       NaN  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       NaN  

[215 rows x 15 columns]

In [5]:
from Univariate import UnivariateBuilder
quan,qual = UnivariateBuilder.quanQualFinder(dataset)
print("quan:",quan)
print("qual:",qual)

quan: ['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']
qual: ['gender', 'ssc_b', 'hsc_b', 'hsc_s', 'degree_t', 'workex', 'specialisation', 'status']


In [23]:
descriptive = pd.DataFrame()
descriptive

Empty DataFrame
Columns: []
Index: []

In [24]:
descriptive = pd.DataFrame(index = ["Mean","Median","Mode"],columns=quan)
descriptive

sl_no ssc_p hsc_p degree_p etest_p mba_p salary
Mean     NaN   NaN   NaN      NaN     NaN   NaN    NaN
Median   NaN   NaN   NaN      NaN     NaN   NaN    NaN
Mode     NaN   NaN   NaN      NaN     NaN   NaN    NaN

In [39]:
for colName in quan:
    descriptive[colName]['Mean'] = dataset[colName].mean()
    descriptive[colName]['Median'] = dataset[colName].median()
    descriptive[colName]['Mode'] = dataset[colName].mode()[0]
    

C:\Users\Admin\AppData\Local\Temp\ipykernel_30400\3147622687.py:2: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

  descriptive[colName]['Mean'] = dataset[colName].mean()
C:\Users\Admin\AppData\Local\Temp\ipykernel_30400\3147622687.py:3: Future

In [40]:
descriptive

sl_no      ssc_p      hsc_p   degree_p    etest_p      mba_p  \
Mean    108.0  67.303395  66.333163  66.370186  72.100558  62.278186   
Median  108.0       67.0       65.0       66.0       71.0       62.0   
Mode        1       62.0       63.0       65.0       60.0       56.7   

               salary  
Mean    288655.405405  
Median       265000.0  
Mode         300000.0

### Understood from POC

##### Mean, Median, Mode values we can find only for interger or float datatype only.

#### Find Informations about it.

#### Mean value:
##### Mean to find average.
The candidate has scored 67% in SSC, 66% in HSC, 66% in their degree, 72% in the entrance test (improved), and 62% in MBA (slightly decreased / average performance).

Note: Don't consider salary related values in mean method becasue it calculated with outliers.

#### Median value:
##### Median NOT allow outliers.
The median salary is 265000.0.

#### Mode value:
##### Mode is used to find repeated values.
The candidate repeatedly scored similar marks throughout their academics: 62 marks in SSC, 63 marks in HSC, 65 marks in their degree, 60 marks in the entrance test, and 56 marks in MBA.
